hop/publish.py

```diff
@@ -4,64 +4,38 @@
 __description__ = "tools to parse and publish messages"
 
 
-import argparse
-import warnings
-
+from .auth import load_auth
 from . import cli
-from .io import Stream
-from .models import GCNCircular, VOEvent, MessageBlob
+from . import io
 
 
 # ------------------------------------------------
 # -- main
 
 
 def _add_parser_args(parser):
-    cli.add_url_opts(parser)
+    cli.add_client_opts(parser)
     parser.add_argument(
         "message", metavar="MESSAGE", nargs="+", help="One or more messages to publish.",
     )
-    cli.add_config_opts(parser)
-
     parser.add_argument(
         "-f",
         "--format",
-        type=str,
-        default="blob",
-        help="Specifies the format of the message(s), such as 'circular' or 'voevent'. "
-        "Specify 'blob' if sending an unstructured message. Default: 'blob'.",
+        choices=io.Deserializer.__members__,
+        default=str(io.Deserializer.BLOB),
```

Comment:
This needs to be `default=str(io.Deserializer.BLOB.name),`, otherwise the default format isn't loaded correctly:
```
#: hop publish kafka://scimma-server:9092/test tests/data/example.gcn3   # don't specify a format, so it should be a BLOB

Traceback (most recent call last):
  File "/opt/conda/bin/hop", line 10, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.7/site-packages/hop/__main__.py", line 72, in main
    args.func(args)
  File "/opt/conda/lib/python3.7/site-packages/hop/publish.py", line 35, in _main
    loader = io.Deserializer[args.format]
  File "/opt/conda/lib/python3.7/enum.py", line 352, in __getitem__
    return cls._member_map_[name]
KeyError: 'Deserializer.BLOB'
```

Reply:
Thanks, this should be fixed now.